### ISSUE TITLE
'Eror: Wildcard selector is not supported' in `useUpdateMutation`

### ISSUE THREAD

**Issue by leeuwd** [psteinroe/supabase-cache-helpers#158]
**Describe the bug**

A mutation hook like this:

```ts
export const useProfilesUpdateMutation = () => {
  const supabase = useSupabaseClient<Database>()
  const table = supabase.from('profiles')

  return useUpdateMutation(table, ['id'])
}
```

Is being consumed like this:

```ts
const { trigger: update } = useProfilesUpdateMutation()
```

And being used like this:

```ts
const data = {
    id: freshProfile.id,
    newsletter: value,
    updated_at: new Date().toISOString(),
}

await update(data)
```

Then, when `update(data)` is called, it yields:

```
Error: Wildcard selector is not supported

 ```

**To Reproduce**

I aim to sandbox the issue soon. 

**Expected behavior**

Mutation is executed.

**Additional context**

Console error:

<img width="696" alt="image" src="https://user-images.githubusercontent.com/1755190/232028736-c8936d6e-5b6e-4031-9646-12aa87676e70.png">

The new docs look nice, it missed some usage examples. I've inferred the usage from the source's tests, but I just might be missing something here.

**Comment by psteinroe** [comment]
Thanks for reporting the issue! This is a limitation of cache helpers: You cannot use the wildcard `*` selector, because we do not know what columns the table has, and thereby cannot derive whether the query cache should be updated with the new input. The issue occurs most likely because you have a `useQuery` (or any pagination query hook) on the `profiles` table somewhere in your app, that selects all columns with `.select('*')`. Can you confirm this?

I was thinking about how to circumvent this, but without explicit knowledge of the tables columns' the cache update logic does not work. 😢 

But this definitely has to be more prominent in the docs. Any suggestions where to put it?

**Comment by leeuwd** [comment]
Thanks for the quick reply (and the great library!).

I've totally missed that the mutations and query are related in that sense. 

As doc improvement, to speak for myself where I would have expected such a note, would be at https://supabase-cache-helpers.vercel.app/postgrest/queries#usequery, as additional notes below the code example.

Perhaps, mention it as inline code doc too?

```ts
import { useQuery } from "@supabase-cache-helpers/postgrest-swr";
 
const client = createClient<Database>(
  process.env.SUPABASE_URL,
  process.env.SUPABASE_ANON_KEY
);
 
function Page() {
  const { data, count } = useQuery(
    client
      .from("contact")
      // We recommend you to be explicit in the select statement fields, and not use a wildcard ('*'), since
      // mutations need to derive whether the query cache should be updated with new values
      .select("id,username,ticket_number", { count: "exact" })
      .eq("username", "psteinroe"),
    {
      revalidateOnFocus: false,
      revalidateOnReconnect: false,
    }
  );
  return <div>...</div>;
}
```

Or, perhaps, even make select values always required?

```ts
...(columns: Query, ....)
```

**Comment by psteinroe** [comment]
Thanks! It's super hard to explain what is going on under the hood. I tried to convey it [here](https://supabase-cache-helpers.vercel.app/postgrest/mutations). Basically, with every mutation, we scan the cache for entries of the same table and extend the `.select()` statement of the mutation with all relevant data, and then pass the fresh data to the cache mutations that populate the data. This keeps your entire app up-to-date without doing anything. You can always opt-out of this behaviour with `disableAutoQuery: true`. 

I improved the types for the mutations to now allow any `*` within the query, but I cannot do the same for queries unfortunately since there is no way to control the type of the query string. 😞 

WDYT of https://github.com/psteinroe/supabase-cache-helpers/pull/159 ?

**Comment by leeuwd** [comment]
LGTM! Would have helped me. 

Thanks for your help and the great lib!

**Comment by psteinroe** [comment]
yes, the cache update works for relationship queries of and level and filters on those the same way it does for normal columns. Hence being explicit is required on every level unfortunately. 😟 

I will add that to the docs, too!

**Comment by psteinroe** [comment]
added the hint to the docs. reopen if you have more ideas for improving the dx or docs here. 🚀
